src/libstd/sys/hermit/net.rs

```diff
@@ -125,42 +141,64 @@ impl TcpStream {
     }
 
     pub fn peer_addr(&self) -> io::Result<SocketAddr> {
-        Err(io::Error::new(ErrorKind::Other, "peer_addr isn't supported"))
+        let (ipaddr, port) = abi::tcpstream::peer_addr(*self.0.as_inner())
+            .map_err(|_| io::Error::new(ErrorKind::Other, "peer_addr failed"))?;
+
+        let saddr = match ipaddr {
+            Ipv4(ref addr) => SocketAddr::new(
+                IpAddr::V4(Ipv4Addr::new(addr.0[0], addr.0[1], addr.0[2], addr.0[3])),
+                port,
+            ),
+            Ipv6(ref addr) => SocketAddr::new(
+                IpAddr::V6(Ipv6Addr::new(
+                    ((addr.0[0] as u16) << 8) | addr.0[1] as u16,
+                    ((addr.0[2] as u16) << 8) | addr.0[3] as u16,
+                    ((addr.0[4] as u16) << 8) | addr.0[5] as u16,
+                    ((addr.0[6] as u16) << 8) | addr.0[7] as u16,
+                    ((addr.0[8] as u16) << 8) | addr.0[9] as u16,
+                    ((addr.0[10] as u16) << 8) | addr.0[11] as u16,
+                    ((addr.0[12] as u16) << 8) | addr.0[13] as u16,
+                    ((addr.0[14] as u16) << 8) | addr.0[15] as u16)),
```

Comment:
```suggestion
                IpAddr::V6(Ipv6Addr::from(addr.0),
```

Reply:
This doesn't work on my system. How do you convert an array of u8 to an army of u16?

Reply:
There is `impl From<[u8; 16]> for Ipv6Addr`: https://doc.rust-lang.org/nightly/std/net/struct.Ipv6Addr.html#impl-From%3C%5Bu8%3B%2016%5D%3E

Reply:
Ok, I will test it. Thanks for the hint!